### DISCUSSION TITLE
Webservices xml - post new item's picture

### DISCUSSION THREAD

**Discussion by jcvoyeux** [PrestaShop/PrestaShop#34754] [Ideas]
### The problem

Hello,
When we post a new picture we receive the picture as answer.
An xml message should be better.
We just need to know if the picture is well inserted or not. We don't need to get back the picture.
A short xml file with just  OK (and the ID of the picture) is smaller and faster.
I know in php you propose to test the return code of the requeste (200) but I'm not in PHP, I use directly curl tool in command line and need a xml answer.

### The solution

Send an xml message with "OK" and/or the picture id if the picture is well inserted/modified.
I think the best webservices are :
"Give me something" -> receive the thing or error
"Do something" -> done (OK with possibly an id) or error (NOK + why)

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
Hi @jcvoyeux 

The return, by default, is just an XML with one of the fields being a base64 encoded image.

**Comment by jcvoyeux** [comment]
I know but I think a short answer with only “ok” would be better and faster to receive.
Receiving the picture that I just sent is not necessary and heavy.

De : Krystian Podemski ***@***.***
Envoyé : jeudi 14 décembre 2023 12:36
À : PrestaShop/PrestaShop ***@***.***>
Cc : JC ***@***.***>; Mention ***@***.***>
Objet : Re: [PrestaShop/PrestaShop] Webservices xml - post new item's picture (Discussion #34754)


Hi @jcvoyeux<https://github.com/jcvoyeux>

The return, by default, is just an XML with one of the fields being a base64 encoded image.
